autonat.go

```diff
@@ -188,7 +188,7 @@ func (as *AmbientAutoNAT) background() {
 			}
 
 		case e := <-subChan:
-			switch e.(type) {
+			switch e := e.(type) {
```

Comment:
is this free? I'd avoided doing it since the typed event wasn't needed in the localAddressesUpdated case.

Reply:
You are correct... by about a half a nanosecond. It saves a copy.

Up to you. I prefer this style because I like to avoid type panicing type assertions where possible.